rules/alerting.go

```diff
@@ -275,7 +305,10 @@ func (r *AlertingRule) Eval(ctx context.Context, ts time.Time, query QueryFunc,
 			a.FiredAt = ts
 		}
 
-		vec = append(vec, r.sample(a, ts))
+		if r.restored {
```

Comment:
We only lose the samples of first 2 `Eval` at startup (because it is restored after 2nd Eval), I guess that's not a big harm? 

In my view if the restoration changed the state of the alert, then adding the samples before restoration can be like noise in the graph.

Reply:
Ah, I see. One sec. But if Group is created with `shouldRestore=false`, then `restored` is never true. Which means we never insert samples?

If `NewGroup` is to be called with `shouldRestore=true` always, remove that parameter?

Reply:
From [this](https://github.com/prometheus/prometheus/pull/4061/files#diff-d23c7264c420b842612a4fe05e018217R677), `m.restored=true` when `shouldRestore=false`.
And we pass `m.restored` to the rule [here](https://github.com/prometheus/prometheus/pull/4061/files#diff-d23c7264c420b842612a4fe05e018217R705).
So if `shouldRestore=false`, `rule.restored=true`, hence samples are added.


And no, when groups are re-loaded using API call, `shouldRestore=false`, hence that parameter would be needed.

Reply:
Ah, I see. Got it.